GliaWidgets/CallVisualizer/VisitorCode/VisitorCodeViewController.swift

```diff
@@ -0,0 +1,103 @@
+import UIKit
+
+final class VisitorCodeViewController: UIViewController {
+
+    var timer: Timer?
+    var expirationDate: Date = Date()
+    let visitorCodeView: VisitorCodeView
+    let theme: Theme
+
+    private let viewType: ViewType
+
+    enum ViewType {
+        case alert
+        case view
+    }
+
+    init(
+        viewType: ViewType,
+        uiConfig: RemoteConfiguration? = nil
+    ) {
+        let theme = Theme()
+        if let uiConfig = uiConfig {
+            theme.applyRemoteConfiguration(uiConfig, assetsBuilder: .standard)
+        }
+        self.theme = Theme()
+        self.viewType = viewType
+        visitorCodeView = VisitorCodeView(with: theme.alert)
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    override func loadView() {
+        switch viewType {
+        case .view:
+            view = visitorCodeView
+        case .alert:
+            view = withContainer(visitorCodeView)
+            view.backgroundColor = .black.withAlphaComponent(0.3)
+            visitorCodeView.addShadow()
+            visitorCodeView.addCloseButton()
+        }
+    }
+
+    private func withContainer(_ view: VisitorCodeView) -> UIView {
+
+        let container = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        container.addSubview(view)
+        NSLayoutConstraint.activate([
+            container.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            container.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            container.widthAnchor.constraint(greaterThanOrEqualTo: view.safeAreaLayoutGuide.widthAnchor, constant: 16)
+        ])
+
+        return container
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        visitorCodeView.title = "Your Visitor Code"
+        visitorCodeView.showsPoweredBy = theme.showsPoweredBy
+        visitorCodeView.closeTapped = dismissAlert
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        timer?.invalidate()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        updateVisitorCode()
+    }
+
+    func updateVisitorCode() {
```

Comment:
would be clearer to name method like scheduleTimer or somehow that will correspond to what method does

Reply:
fixed [fc50a16](https://github.com/salemove/ios-sdk-widgets/pull/448/commits/fc50a168934e072d2a4b217c22bcbdc235a31d95)